presto-accumulo/src/main/java/com/facebook/presto/accumulo/io/AccumuloRecordCursor.java

```diff
@@ -0,0 +1,584 @@
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.FirstEntryInRowIterator;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.io.Text;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.facebook.presto.accumulo.AccumuloErrorCode.INTERNAL_ERROR;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.IO_ERROR;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.TimeType.TIME;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Implementation of Presto RecordCursor, responsible for iterating over a Presto split, reading
+ * rows of data and then implementing various methods to retrieve columns within each row.
+ *
+ * @see AccumuloRecordSet
+ * @see AccumuloRecordSetProvider
+ */
+public class AccumuloRecordCursor
+        implements RecordCursor
+{
+    private final List<AccumuloColumnHandle> cHandles;
+    private final String[] fieldToColumnName;
+
+    private long bytesRead = 0L;
+    private long nanoStart = 0L;
+    private long nanoEnd = 0L;
+    private final BatchScanner scan;
+    private final Iterator<Entry<Key, Value>> iterator;
+    private final AccumuloRowSerializer serializer;
+    private Entry<Key, Value> prevKV;
+    private Text prevRowID = new Text();
+    private Text rowID = new Text();
+
+    /**
+     * Creates a new instance of {@link AccumuloRecordCursor}
+     *
+     * @param serializer Serializer to decode the data stored in Accumulo
+     * @param scan BatchScanner for retrieving rows of data
+     * @param rowIdName Presto column that is the Accumulo row ID
+     * @param cHandles List of column handles in each row
+     * @param constraints List of all column constraints
+     */
+    public AccumuloRecordCursor(AccumuloRowSerializer serializer, BatchScanner scan,
+            String rowIdName, List<AccumuloColumnHandle> cHandles,
+            List<AccumuloColumnConstraint> constraints)
+    {
+        this.cHandles = requireNonNull(cHandles, "cHandles is null");
+        this.scan = requireNonNull(scan, "scan is null");
+
+        this.serializer = requireNonNull(serializer, "serializer is null");
+        this.serializer.setRowIdName(rowIdName);
+
+        // If there are no columns, or the only column is the row ID, then
+        // configure a scan iterator to only return the row IDs
+        if (cHandles.size() == 0
```

Comment:
Can you extract this condition to a separate method to make it more meaningful and be able to get rid of the comment above?